provider/slof.py

```diff
@@ -0,0 +1,192 @@
+"""
+Module for providing common interface to test SLOF component.
+"""
+
+import re
+import time
+import logging
+import os
+
+from virttest import utils_misc
+
+
+class _LazyReader(object):
+    """
+    Custom a file container(iterable) for reading SLOF contents.
+    """
+    def __init__(self, vm):
+        self._filename = vm.serial_console_log
+        self._fd = open(self._filename)
+        self._len = self._get_len()
+
+    def __iter__(self):
+        pos = 0
+        with open(self._filename) as f:
+            for line in f:
+                yield line, pos
+                pos += 1
+
+    def _get_len(self):
+        num = 0
+        while 1:
+            try:
+                next(self._fd)
+            except StopIteration:
+                return num
+            num += 1
+
+    def __getitem__(self, index):
+        if isinstance(index, slice):
+            start, stop, step = index.indices(self._len)
+            for line, pos in self:
+                if start <= pos <= stop:
+                    yield line, pos
+        else:
+            raise TypeError('Index must be class slice.')
+
+
+def get_boot_content(vm, start_pos=0, start_str='SLOF',
```

Comment:
could use a global dictionary to store reader object, no need to instantiate for each call to `get_boot_content`.
```python
reader_map = {}
def get_boot_content(vm, start_pos, start_str', end_str):
    content = []
    reader = reader_map.setdefault(vm.name, _LazyReader(vm))
    ...
```